### DISCUSSION TITLE
pom文件中下载不到的jar包和上下不对应的版本号应该怎么处理呢？

### DISCUSSION THREAD

**Discussion by hhh220** [apolloconfig/apollo#4043] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/92715270/137691042-ac2c1895-ce5a-403b-925a-cb949596ea17.png)

![image](https://user-images.githubusercontent.com/92715270/137691158-b312c296-c138-4a2c-a824-e583bbe2d02a.png)

**Comment by nobodyiam** [comment, marked as answer]
`env.GITHUB_OAUTH_TOKEN` is a variable set by travis ci, though I think it may not be used anymore.
[maven-checkstyle-plugin](https://search.maven.org/artifact/org.apache.maven.plugins/maven-checkstyle-plugin/2.17/maven-plugin) and [maven-pmd-plugin](https://search.maven.org/artifact/org.apache.maven.plugins/maven-pmd-plugin/3.6/maven-plugin) are only used in reporting process, which won't affect the normal use. I checked the version that they did exist in maven central, so I'm not sure whether they are not found..

**Reply by hhh220** [reply]
Thank you，I get。I have an other question——where is auth configuration file in portal service？
![image](https://user-images.githubusercontent.com/92715270/138051331-106162fd-63f5-4fd9-9750-0026b22c41b3.png)

**Reply by hhh220** [reply]
Is it in SpringSecurityAuthAutoConfiguration of AuthConfiguration?
![image](https://user-images.githubusercontent.com/92715270/138054287-db990664-10c2-4403-abb4-49fe3be57379.png)

**Reply by nobodyiam** [reply]
Yes, that's the one.

**Reply by hhh220** [reply]
3q

**Reply by hhh220** [reply]
this is another question—what‘s the meaning of this line of comment？
![image](https://user-images.githubusercontent.com/92715270/138240523-3a593027-454a-4be3-abf0-dc30574b2a73.png)

**Reply by hhh220** [reply]
Why the cluster name might be overridden，for example？

**Reply by nobodyiam** [reply]
e.g. when the cluster name passed in is `test` but the actual cluster name of the release is `default`.

**Reply by hhh220** [reply]
i get it,3q

**Reply by hhh220** [reply]
another question—
How to set the priority of local application.properties over the configuration of Apollo？

**Reply by hhh220** [reply]
e.g.   I want to use local  sever.port=8081  to cover  sever.port=8085 on Apollo?

**Reply by hhh220** [reply]
which method can be effective excepting that Apollo is not configured the sever.port？

**Reply by Anilople** [reply]
-Dserver.port=xxx?

**Reply by nobodyiam** [reply]
System properties and OS Environment variables are preferred, as is mentioned by @Anilople .
Meanwhile, you may also take a look at #1560.

**Reply by hhh220** [reply]
port is an example. I have some configurations like this ，so system properties and os environment are not  available。The same configurations between local and apollo I need  local to cover apollo。

**Reply by hhh220** [reply]
I try it 1560 but failed。
![image](https://user-images.githubusercontent.com/92715270/138628123-9e3f6899-e23c-44c1-9c69-8efc46d501d6.png)

![image](https://user-images.githubusercontent.com/92715270/138628071-77756825-f429-4f19-9759-f9567a698e8a.png)

![image](https://user-images.githubusercontent.com/92715270/138628429-7e8bc06b-71e7-4b2f-8334-4421c35c04f0.png)

**Reply by hhh220** [reply]
update the last picture
![image](https://user-images.githubusercontent.com/92715270/138628597-f9e22381-cf63-484f-8ebb-7462d43cfeb7.png)

**Reply by hhh220** [reply]
this is the configuration of my application.properties.
![image](https://user-images.githubusercontent.com/92715270/138644303-725611db-e7c9-42d8-a536-b2d8b5d9cea7.png)

**Reply by hhh220** [reply]
It doesn't work to put TEST1.apolloclient-local.properties directly under META-INF.

**Reply by hhh220** [reply]
It worked. I missed the config package，it’s my fault。I‘m sorry，3q.

**Reply by hhh220** [reply]
I find a new question—when I delete some apollo configuration，the result of getting from @Value is old value . However,update some  apollo configuration，the result of getting from @Value is new value .Am I need to configure a listener to update value？
